File: app/src/main/java/at/shockbytes/dante/util/books/BookConfig.java
package at.shockbytes.dante.util.books;

import io.realm.RealmObject;

/**
 * @author Martin Macheiner
 *         Date: 28.08.2016.
 */
public class BookConfig extends RealmObject {

    private long lastPrimaryKey;

    public BookConfig() {
        this(0);
    }

    public BookConfig(long lastPrimaryKey) {
        this.lastPrimaryKey = lastPrimaryKey;
    }

    public long getLastPrimaryKey() {
        long key = lastPrimaryKey;
        lastPrimaryKey ++;
        return key;
    }

}

File: app/src/main/java/at/shockbytes/dante/util/books/BookConfig.kt
package at.shockbytes.dante.util.books

import io.realm.RealmObject

/**
 * @author Martin Macheiner
 * Date: 28.08.2016.
 */
open class BookConfig(private var lastPrimaryKey: Long = 0) : RealmObject() {

    fun getLastPrimaryKey(): Long {
        val key = lastPrimaryKey
        lastPrimaryKey++
        return key
    }

}
